src/unix/linux/disk.rs

```diff
@@ -173,63 +177,96 @@ fn get_actual_device_name(device: &OsStr) -> String {
+) -> Option<(u64, u64, bool)> {
+    if refresh_kind.total_space() || refresh_kind.available_space() || refresh_kind.is_read_only() {
+        let mount_point_cpath = to_cpath(mount_point);
         let mut stat: statvfs = mem::zeroed();
         if retry_eintr!(statvfs(mount_point_cpath.as_ptr() as *const _, &mut stat)) == 0 {
             let bsize = cast!(stat.f_bsize);
             let blocks = cast!(stat.f_blocks);
             let bavail = cast!(stat.f_bavail);
-            total = bsize.saturating_mul(blocks);
-            available = bsize.saturating_mul(bavail);
-            is_read_only = (stat.f_flag & libc::ST_RDONLY) != 0;
-        }
-        if total == 0 {
-            return None;
+            let total = bsize.saturating_mul(blocks);
+            let available = bsize.saturating_mul(bavail);
+            let is_read_only = (stat.f_flag & libc::ST_RDONLY) != 0;
+
+            if total == 0 {
+                None
+            } else {
+                Some((total, available, is_read_only))
+            }
+        } else {
+            None
         }
-        let mount_point = mount_point.to_owned();
-        let is_removable = removable_entries
-            .iter()
-            .any(|e| e.as_os_str() == device_name);
-
-        let actual_device_name = get_actual_device_name(device_name);
-
-        let (read_bytes, written_bytes) = procfs_disk_stats
-            .get(&actual_device_name)
-            .map(|stat| (stat.sectors_read * SECTOR_SIZE, stat.sectors_written * SECTOR_SIZE))
-            .unwrap_or_default();
-
-        Some(Disk {
-            inner: DiskInner {
-                type_,
-                device_name: device_name.to_owned(),
-                actual_device_name,
-                file_system: file_system.to_owned(),
-                mount_point,
-                total_space: cast!(total),
-                available_space: cast!(available),
-                is_removable,
-                is_read_only,
-                old_read_bytes: 0,
-                old_written_bytes: 0,
-                read_bytes,
-                written_bytes,
-            },
-        })
+    } else {
+        Some((Default::default(), Default::default(), Default::default()))
     }
 }
 
+fn new_disk(
+    device_name: &OsStr,
+    mount_point: &Path,
+    file_system: &OsStr,
+    removable_entries: &[PathBuf],
+    procfs_disk_stats: &HashMap<String, DiskStat>,
+    refresh_kind: DiskRefreshKind,
+) -> Option<Disk> {
+    let type_ = if refresh_kind.kind() {
+        find_type_for_device_name(device_name)
+    } else {
+        // TODO: discuss the representation of "you opted out of refreshing the DiskKind"
+        DiskKind::Unknown(-1)
+    };
+
+    let (total_space, available_space, is_read_only) = match unsafe { load_statvfs_values(mount_point, refresh_kind) } {
+        Some((total_space, available_space, is_read_only)) => (total_space, available_space, is_read_only),
+        None => return None,
```

Comment:
Should likely not return here.

Also, should be behind a `if`.

Reply:
The previous behavior was to return `None` if the call to `statvfs` failed. Do you want me to update it to quietly not populate the `statvfs` fields, instead?

Reply:
Yes please.

Reply:
Done.